test/Land/Model/freeze_thaw_alone.jl

```diff
@@ -0,0 +1,165 @@
+# Test that freeze thaw alone reproduces expected behavior: exponential behavior
+# for liquid water content, ice content, and total water conserved
+
+using MPI
+using OrderedCollections
+using StaticArrays
+using Statistics
+using Test
+
+using CLIMAParameters
+struct EarthParameterSet <: AbstractEarthParameterSet end
+const param_set = EarthParameterSet()
+using CLIMAParameters.Planet: ρ_cloud_liq
+using CLIMAParameters.Planet: ρ_cloud_ice
+
+using ClimateMachine
+using ClimateMachine.Land
+using ClimateMachine.Land.SoilWaterParameterizations
+using ClimateMachine.Land.SoilHeatParameterizations
+using ClimateMachine.Mesh.Topologies
+using ClimateMachine.Mesh.Grids
+using ClimateMachine.DGMethods
+using ClimateMachine.DGMethods.NumericalFluxes
+using ClimateMachine.DGMethods: BalanceLaw, LocalGeometry
+using ClimateMachine.MPIStateArrays
+using ClimateMachine.GenericCallbacks
+using ClimateMachine.ODESolvers
+using ClimateMachine.VariableTemplates
+using ClimateMachine.SingleStackUtils
+using ClimateMachine.BalanceLaws:
+    BalanceLaw, Prognostic, Auxiliary, Gradient, GradientFlux, vars_state
+
+@testset "Freeze thaw alone" begin
+    FT = Float64
+    struct tmp_model <: BalanceLaw end
+    struct tmp_param_set <: AbstractParameterSet end
+
+    function get_grid_spacing(
+        N_poly::Int64,
+        nelem_vert::Int64,
+        zmax::FT,
+        zmin::FT,
+    )
+        test_config = ClimateMachine.SingleStackConfiguration(
+            "TmpModel",
+            N_poly,
+            nelem_vert,
+            zmax,
+            tmp_param_set(),
+            tmp_model();
+            zmin = zmin,
+        )
+
+        Δ = min_node_distance(test_config.grid)
+        return Δ
+    end
+
+    function init_soil_water!(land, state, aux, coordinates, time)
+        myf = eltype(state)
+        state.soil.water.ϑ_l = myf(land.soil.water.initialϑ_l(aux))
+        state.soil.water.θ_i = myf(land.soil.water.initialθ_i(aux))
+    end
```

Comment:
```suggestion

function get_grid_spacing(
    N_poly::Int64,
    nelem_vert::Int64,
    zmax::FT,
    zmin::FT,
) where {FT}
    test_config = ClimateMachine.SingleStackConfiguration(
        "TmpModel",
        N_poly,
        nelem_vert,
        zmax,
        tmp_param_set(),
        tmp_model();
        zmin = zmin,
    )

    Δ = min_node_distance(test_config.grid)
    return Δ
end

function init_soil_water!(land, state, aux, coordinates, time)
    FT = eltype(state)
    state.soil.water.ϑ_l = FT(land.soil.water.initialϑ_l(aux))
    state.soil.water.θ_i = FT(land.soil.water.initialθ_i(aux))
end

@testset "Freeze thaw alone" begin
    FT = Float64
    struct tmp_model <: BalanceLaw end
    struct tmp_param_set <: AbstractParameterSet end

```

Reply:
Not that it's a performance problem, but there is a closure over `FT` in `get_grid_spacing`.

Reply:
thank you! will fix now

Reply:
I thought 
```function init_soil_water!(land, state, aux, coordinates, time)
    FT = eltype(state)
    state.soil.water.ϑ_l = FT(land.soil.water.initialϑ_l(aux))
    state.soil.water.θ_i = FT(land.soil.water.initialθ_i(aux))
end```
was also an issue, but just made it eltype(state)(land.soil.water.initial....) now, instead of declaring a new variable.